File: game/Assets/Standard Assets/Character Controllers/Sources/Scripts/Hatch.js
#pragma strict

// var pivot : Transform;
// // this is half the length of the side of the cube
// var halfSideLength : float = 1;
// private var isRotating : boolean = false;
// function Start(){
// //set the pivot to be in the plane the cube is on
// if (!pivot) {
// 	Debug.Log("Assigning pivot variable...");
// 	pivot = transform;
// };
// pivot.position.y = transform.position.y - halfSideLength;
// }
// function Update () {
 
//       if(isRotating == false){
//               rotateRight(360,1);    }
 
// }
// function rotateRight(degrees : float, time : float) {
 
//    var speed = 1.0/time;
//    var initialRotation = Quaternion.identity;
//    pivot.position.x = transform.position.x + halfSideLength;
//    transform.parent = pivot;
//  //  pivot.localEulerAngles.z = -degrees;
//  //  var finalRotation = pivot.transform.rotation;
//    isRotating = true;
//    for (var t = 0.0; t < 1.0; t += Time.deltaTime*speed){
//       //pivot.transform.rotation = Quaternion.Slerp(initialRotation,finalRotation, t);
//       pivot.transform.eulerAngles.z = t*(-degrees); //Mathf.Lerp(0,-degrees,t);
//       yield;
//    }
//    pivot.localEulerAngles.z = -degrees;
//    transform.parent = null;
//    pivot.rotation = Quaternion.identity;
//    isRotating = false;
// }

function Update() {
		
}

function Start () {
	var vertices = GetComponent(MeshFilter).mesh.vertices;
	for (var i = vertices.length - 1; i >= 0; i--) {
		Debug.Log(i + "th:");
		Debug.Log(vertices[i]);
	};
}

File: game/Assets/Standard Assets/Character Controllers/Sources/Scripts/Hatch.cs
using UnityEngine;
using System.Collections;

public class Hatch : MonoBehaviour {

	public Transform pivot;
	// this is half the length of the side of the cube
	float halfSideLength = 1.0f;
	bool isRotating = false;

	void OnEnable() {
		EventManager.OnAction += Open;
		Debug.Log(GetComponent<MeshFilter>().mesh.bounds);

	}
	void OnDisable() {
		EventManager.OnAction -= Open;
	}

	public void Open() {
		var pos = transform.position;
		pos.x += Random.Range(-10.0f, 30.0f);
		transform.position = pos;
	}

	// Use this for initialization
	void Start () {
		pivot = transform;
		var pos = pivot.position;
		pos.y = transform.position.y - halfSideLength;
		pivot.position = pos;
	}
		
	public IEnumerator rotateRight(float degrees, float time) {
		var speed = 1.0f/time;
		var initialRotation = Quaternion.identity;
		var pos = pivot.position;
		pos.x = transform.position.x + halfSideLength;
		pivot.position = pos;
		transform.parent = pivot;
		pivot.rotation = Quaternion.identity;
		isRotating = true; 
		Debug.Log("Hey....");
		for (float t = 0.0f; t < 1.0f; t += Time.deltaTime*speed){
			var tmp = pivot.transform.eulerAngles;
			tmp.z = t*(-degrees);
			pivot.transform.eulerAngles = tmp;
			yield return null;
		}
		var lea = pivot.localEulerAngles;
		transform.parent = null;
		lea.z = -degrees;
		pivot.localEulerAngles = lea;
		isRotating = false; 
	} 


	// Update is called once per frame
	void Update () {
		if (!isRotating) {
			Debug.Log("Rotating...");
			rotateRight(90,1);
		}
	}
}
